xbmc/epg/Epg.cpp

```diff
@@ -52,9 +50,7 @@ CEpg::CEpg(int iEpgID, const CStdString &strName /* = "" */, const CStdString &s
   m_pvrChannel = empty;
 }
 
-CEpg::CEpg(CPVRChannelPtr channel, bool bLoadedFromDb /* = false */) :
-    m_bChanged(!bLoadedFromDb),
-    m_bTagsChanged(false),
+CEpg::CEpg(CPVRChannelPtr channel) :
```

Comment:
Well, no. :-) I don't get you point, sorry. I don't say that using the "per value" approach does not work, but it is inefficient, as a (in my opinion) unnecessary copy of the shared_ptr instance is made. What does not work with the const ref approach? Why does it violate "the concept?

shared_ptr::swap does not work here, as no transfer of ownership of the pointer is intended.